src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -1659,7 +1662,19 @@ export class SearchView extends ViewPane {
 			if (typeof node === 'string') {
 				dom.append(span, document.createTextNode(node));
 			} else {
-				const link = this.instantiationService.createInstance(Link, node);
+				const link = this.instantiationService.createInstance(Link, node, {
+					opener: async href => {
+						const parsed = URI.parse(href, true);
+						if (parsed.scheme === Schemas.command) {
+							const result = await this.commandService.executeCommand(parsed.path);
```

Comment:
We don't have to solve this now but something that I would think about at some point is whether extensions should have to opt the message in to this feature. For example an extension has to set `isTrusted` on a `MarkdownString` for command links to work. If an extension is blindly sending back messages that came from some 3rd party search tool, they might not want command links to be injected. And you should look at where these messages come from in your search scenarios and make sure that won't be an issue.